src/verify.rs

```diff
@@ -197,6 +202,64 @@ pub fn verify_server_cert(roots: &RootCertStore,
     .map_err(|err| TLSError::WebPKIError(err))
 }
 
+pub fn parallel_verify_server_cert(roots: &RootCertStore,
+                                   presented_certs: &Vec<ASN1Cert>,
+                                   dns_name: &str) -> Result<(), TLSError> {
+  crossbeam::scope(|scope| {
+    scope.spawn(move || {
+      parallel_verify_common_cert(roots, presented_certs);
+    });
+    let r = scope.spawn(move || {
+      let cert_der = untrusted::Input::from(&presented_certs[0].0);
+      let cert = try!(
+        webpki::EndEntityCert::from(cert_der)
+          .map_err(|err| TLSError::WebPKIError(err))
+      );
+      cert.verify_is_valid_for_dns_name(untrusted::Input::from(dns_name.as_bytes()))
+      .map_err(|err| TLSError::WebPKIError(err))
+    });
+    r.join()
+  })
+  
+}
+
+fn parallel_verify_common_cert<'a>(roots: &RootCertStore,
```

Comment:
Well, it's just my two cents -- I am also just a curious wanna-be contributer:

I think it's probably better not to duplicate such a critical part of the code, so we might be better off find solution that is flexible in whether processing it in parallel or not.
I have no idea how @ctz thinks about adding new dependencies.